### ISSUE TITLE
model loading error

### ISSUE THREAD

**Issue by sathish4ds** [iCVTEAM/PGNet#3]
HI team,

Thanks for the checkpoint I have saved the folder PGNet_DUT%2BHR but getting the following error

![image](https://user-images.githubusercontent.com/69101781/165243283-112cb4ab-d4bd-4052-92ee-6e4d467738bf.png)

**Comment by xiechenxi99** [comment]
Hi, maybe you need to rename the saved model name or the model path in the code.

**Comment by sathish4ds** [comment]
In the given models check point folder we have a data.pkl , version and data folder in the which the checkpoints are there how to get this model-27 file

**Comment by xiechenxi99** [comment]
https://drive.google.com/drive/folders/1hXwCvrdmvkaRePXWPTw5tjFXmrrzHPtt?usp=sharing The downloaded file name should be 'PGNet-DUT-HR-model31',. Put the checkpoint under the PGNet_DUT+HR, and modify the 'model-27' to the right checkpoint name in test.py.

**Comment by domef** [comment]
I'm having a similar problem, I don't understand how to load the downloaded checkpoint. Even if I rename the file, the `PGNet` class is loading 2 checkpoints:

```python
self.resnet = resnet18()
self.swin = Swintransformer(224)
self.swin.load_state_dict(torch.load('../pre/swin224.pth')['model'],strict=False)
self.resnet.load_state_dict(torch.load('../pre/resnet18.pth'),strict=False)
```
 Edit:
It seems that the code that loads the downloaded checkpoint is this (this means that 2 checkpoints are missing):
```python
if self.cfg is not None and self.cfg.snapshot:
    print('load checkpoint')
    pretrain=torch.load(self.cfg.snapshot)
    new_state_dict = {}
    for k,v in pretrain.items():
        new_state_dict[k[7:]] = v  
    self.load_state_dict(new_state_dict, strict=False)  
```

**Comment by sathish4ds** [comment]
hi team,

I have copied a checkpoint of 9mb from data folder in the downloaded zip to PGNet_DUT+2BHR folder but getting this error
![image](https://user-images.githubusercontent.com/69101781/165446631-e7c5ddf9-7ffc-48e8-a352-24a6d120ad8d.png)

**Comment by xiechenxi99** [comment]
> I'm having a similar problem, I don't understand how to load the downloaded checkpoint. Even if I rename the file, the `PGNet` class is loading 2 checkpoints:
> 
> ```python
> self.resnet = resnet18()
> self.swin = Swintransformer(224)
> self.swin.load_state_dict(torch.load('../pre/swin224.pth')['model'],strict=False)
> self.resnet.load_state_dict(torch.load('../pre/resnet18.pth'),strict=False)
> ```
> 
> Edit: It seems that the code that loads the downloaded checkpoint is this (this means that 2 checkpoints are missing):
> 
> ```python
> if self.cfg is not None and self.cfg.snapshot:
>     print('load checkpoint')
>     pretrain=torch.load(self.cfg.snapshot)
>     new_state_dict = {}
>     for k,v in pretrain.items():
>         new_state_dict[k[7:]] = v  
>     self.load_state_dict(new_state_dict, strict=False)  
> ```

Hi, the swin224.pth and resnet18.pth is the pretrained backbone, which is using in train phase. You can simply delete these two lines of code when you test. 
```python
self.swin.load_state_dict(torch.load('../pre/swin224.pth')['model'],strict=False)
self.resnet.load_state_dict(torch.load('../pre/resnet18.pth'),strict=False)
 ```

**Comment by domef** [comment]
Thank you, now it works.

**Comment by xiechenxi99** [comment]
> hi team,
> 
> I have copied a checkpoint of 9mb from data folder in the downloaded zip to PGNet_DUT+2BHR folder but getting this error ![image](https://user-images.githubusercontent.com/69101781/165446631-e7c5ddf9-7ffc-48e8-a352-24a6d120ad8d.png)

Hi, 
Why the checkpoint is 9mb, the file size in the [google drive](https://drive.google.com/drive/folders/1hXwCvrdmvkaRePXWPTw5tjFXmrrzHPtt?usp=sharing ) should be 279.18mb. Please check whether you are using the wrong download link.

**Comment by sathish4ds** [comment]
Thanks now the code is running without any error. I have copied the test images to the data folder but unable to find the result images in the result folder

**Comment by sathish4ds** [comment]
Hi Team do I have any update why the image prediction in not happening

**Comment by xiechenxi99** [comment]
Please check whether the dataset path in test.py is right and the data folder format should be like:

-- data (train dataset and test dataset)
   |-- DAVIS-S
   |   |-- image
   |   |   |-- XXX.jpg
            ...
   |   |-- mask
   |   |   |-- XXX.png
            ...

**Comment by Sharifmhamza** [comment]
I am training the model on a different dataset. I have changed the dataset.py accordingly. But when I am training it from the starch using the pre-trained weights of Swin224 and resnet18, I get an error of args. checkpoint. This parameter is not passing as an argument, but it is mentioned in the code. Which checkpoint should I have to add there? What do you do in that case?

**Comment by xiechenxi99** [comment]
> I am training the model on a different dataset. I have changed the dataset.py accordingly. But when I am training it from the starch using the pre-trained weights of Swin224 and resnet18, I get an error of args. checkpoint. This parameter is not passing as an argument, but it is mentioned in the code. Which checkpoint should I have to add there? What do you do in that case?

Hi, can you provide the specific error messages or screenshots?

**Comment by Sharifmhamza** [comment]
> > I am training the model on a different dataset. I have changed the dataset.py accordingly. But when I am training it from the starch using the pre-trained weights of Swin224 and resnet18, I get an error of args. checkpoint. This parameter is not passing as an argument, but it is mentioned in the code. Which checkpoint should I have to add there? What do you do in that case?
> 
> Hi, can you provide the specific error messages or screenshots?

<img width="1353" alt="Screen Shot 2022-05-27 at 5 38 50 PM" src="https://user-images.githubusercontent.com/32780776/170711167-53e4649a-3051-4949-b062-ec978956f4a6.png">
@xiechenxi99 can you please update me on this issue?

**Comment by xiechenxi99** [comment]
> > > I am training the model on a different dataset. I have changed the dataset.py accordingly. But when I am training it from the starch using the pre-trained weights of Swin224 and resnet18, I get an error of args. checkpoint. This parameter is not passing as an argument, but it is mentioned in the code. Which checkpoint should I have to add there? What do you do in that case?
> > 
> > 
> > Hi, can you provide the specific error messages or screenshots?
> 
> <img alt="Screen Shot 2022-05-27 at 5 38 50 PM" width="1353" src="https://user-images.githubusercontent.com/32780776/170711167-53e4649a-3051-4949-b062-ec978956f4a6.png">
> 
> @xiechenxi99 can you please update me on this issue?

`cfg = Dataset.Config(datapath=args.datapath, savepath=args.savepath,mode='train', batch=args.batchsize, lr=0.03, momen=0.9,
                         decay=5e-4, epoch=32, snapshot=args.checkpoint)`
please try change the args.checkpoint to None

**Comment by Sharifmhamza** [comment]
Thank you issue is resolved.
